android/sharedCode/src/main/java/com/viro/core/ClickListener.kt

```diff
@@ -20,42 +20,39 @@
 //  CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 //  TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 //  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-
-package com.viro.core;
+package com.viro.core
 
 /**
  * Callback interface for responding to click events, which occur when any Controller button is
  * clicked.
  */
-public interface ClickListener {
-
-
+interface ClickListener {
     /**
-     * Callback when a click event is registered over the given {@link Node}.
+     * Callback when a click event is registered over the given [Node].
      *
      * @param source     The platform specific source ID, which indicates what button or
      *                   component on the Controller triggered the event. See the Controller's
      *                   Guide for information.
-     * @param node       The {@link Node} that was clicked.
+     * @param node       The [Node] that was clicked.
      * @param location   The location of the event in world coordinates.
      */
-    void onClick(int source, Node node, Vector location);
+    fun onClick(source: Int, node: Node?, location: Vector?)
 
     /**
-     * Callback when the {@link ClickState} is changed over the given {@link Node}. This callback is
+     * Callback when the [ClickState] is changed over the given [Node]. This callback is
      * for receiving fine-grained information about a click: when the pointer goes down, when it
      * goes up, and if/when the 'click' itself is registered. For a completed click, this callback
-     * is invoked <i>three</i> times: on {@link ClickState#CLICK_DOWN}, {@link ClickState#CLICK_UP},
-     * and then for {@link ClickState#CLICKED}.
-     * <p>
-     * To simply listen for click events, use {@link #onClick(int, Node, Vector)}.
+     * is invoked *three* times: on [ClickState.CLICK_DOWN], [ClickState.CLICK_UP],
+     * and then for [ClickState.CLICKED].
+     *
+     * To simply listen for click events, use [onClick].
      *
      * @param source     The platform specific source ID, which indicates what button or component
      *                   on the Controller triggered the event. See the Controller's Guide for
      *                   information.
-     * @param node       The {@link Node} that was clicked.
+     * @param node       The [Node] that was clicked.
      * @param clickState The status of the click event.
      * @param location   The location of the event in world coordinates.
      */
-    void onClickState(int source, Node node, ClickState clickState, Vector location);
+    fun onClickState(source: Int, node: Node?, clickState: ClickState, location: Vector?)
```

Comment:
@nachtien I think that `node` itself **can't** be null. 

> Node can be null

This two makes sense since not every `id` identify a `node` and a `node` can be already removed.

> location can be null

This also make sense since we can pass a "wrong" location.

Would be possible some logging on this cases?

Reply:
@nachtien Node isn't nullable. But location is nullable I think.

Reply:
Let's just add Kotlin to Gradle for now, I'll convert the other classes first to be sure :)